src/orion/core/utils/tests.py

```diff
@@ -0,0 +1,212 @@
+
+    it swaps the singleton with none at startup and restore them after the tests.
+    It also initialize PickleDB as the storage for testing.
+    We use PickledDB as our storage mock
+
+    Parameters
+    ----------
+    config: YAML
+        YAML config to apply for this test
+
+    experiments: list
+        List of experiments to insert into the database
+
+    trials: list
+        List of trials to insert into the database
+
+    workers: list
+        List of workers to insert into the database
+
+    resources: list
+        List of resources to insert into the database
+
+    database: dict
+        Configuration of the underlying database
+
+    Examples
+    --------
+    >>> myconfig = {...}
+    >>> with OrionTestState(myconfig):
+        ...
+
+    """
+
+    # TODO: Fix these singletons to remove Legacy, MongoDB, PickledDB and EphemeralDB.
+    SINGLETONS = (Storage, Legacy, Database, MongoDB, PickledDB, EphemeralDB)
+    singletons = {}
+    database = None
+    experiments = []
+    trials = []
+    resources = []
+    workers = []
+    tempfile = None
+
+    def __init__(self, experiments=None, trials=None, workers=None, resources=None,
+                 from_yaml=None, database=None):
+        if from_yaml is not None:
+            with open(from_yaml) as f:
+                exp_config = list(yaml.safe_load_all(f))
+                experiments = exp_config[0]
+                trials = exp_config[1]
+
+        self.database_config = _select(database, _get_default_test_database())
+        self.experiments = _select(experiments, [])
+        self.trials = _select(trials, [])
+        self.workers = _select(workers, [])
+        self.resources = _select(resources, [])
+
+    def init(self):
+        """Initialize environment before testing"""
+        self.storage()
+        self.database = get_storage()._db
+        self.load_experience_configuration()
+        return self
+
+    def get_experiment(self, name, user=None, version=None):
+        """Make experiment id deterministic"""
+        exp = Experiment(name, user=user, version=version)
+        exp._id = name
+        return exp
+
+    def get_trial(self, index):
+        """Return a Trial"""
+        return Trial(**self.trials[index])
+
+    def cleanup(self):
+        """Cleanup after testing"""
+        _remove(self.tempfile)
+
+    def load_experience_configuration(self):
+        """Load an example database."""
+        for i, t_dict in enumerate(self.trials):
+            self.trials[i] = Trial(**t_dict).to_dict()
+
+        # sort by submit time like the protocol does
+        self.trials.sort(key=lambda obj: int(obj['_id'], 16), reverse=True)
+
+        for i, _ in enumerate(self.experiments):
+            path = os.path.join(
+                os.path.dirname(__file__),
+                self.experiments[i]["metadata"]["user_script"])
+
+            self.experiments[i]["metadata"]["user_script"] = path
+            self.experiments[i]['version'] = 1
```

Comment:
Why set their version to one? If an experiment is a child of another, this will break.

Reply:
no particular reason, this code essentially adapts:
https://github.com/Epistimio/orion/blob/d23e539d3556c3dea4f6157cf75a528c5ba39125/tests/conftest.py#L150

So I should remove the version or set it to 0 ?